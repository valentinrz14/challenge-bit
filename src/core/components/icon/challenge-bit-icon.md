```TSX
import React, { FunctionComponent } from 'react';
import { ChallengeBitIcon } from './index.tsx';
import { Colors } from '@core/styles/colors';

export const App: FunctionComponent = () => ( 
    <ChallengeBitIcon
        iconProps={{ name: 'home', size: 20, color: Colors.black }}
      />
)
```
