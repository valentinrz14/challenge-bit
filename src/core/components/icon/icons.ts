export { default as Home } from '@assets/svg/home.svg';
export { default as Send } from '@assets/svg/send.svg';
export { default as Error } from '@assets/svg/error.svg';
export { default as Check } from '@assets/svg/check.svg';
export { default as QRCode } from '@assets/svg/qr-code.svg';
export { default as Dollar } from '@assets/svg/dollar.svg';
