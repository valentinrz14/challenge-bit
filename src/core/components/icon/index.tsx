import React, { FunctionComponent } from 'react';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';

/*
 ** Types
 */

interface ChallengeBitIconProps {
  iconProps: {
    name: string;
    size: number;
    color: string;
  };
}

/*
 ** Component
 */

export const ChallengeBitIcon: FunctionComponent<ChallengeBitIconProps> = ({
  iconProps: { name, size, color },
}) => {
  return <Icon name={name} size={size} color={color} />;
};
