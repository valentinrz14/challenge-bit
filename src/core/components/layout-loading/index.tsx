import { Colors } from '@core/styles/colors';
import React, { FunctionComponent } from 'react';
import { ActivityIndicator, StyleSheet, View } from 'react-native';

/*
 ** Types
 */
interface LayoutLoadingProps {
  color?: string;
  size?: 'large' | 'small';
}
/*
 ** Styles
 */

const styles = StyleSheet.create({
  wrapperLayoutLoading: {
    marginTop: 20,
    alignItems: 'center',
    justifyContent: 'center',
  },
});
/*
 ** Component
 */
export const LayoutLoading: FunctionComponent<LayoutLoadingProps> = ({
  color = Colors.blue,
  size = 'large',
}) => {
  return (
    <View style={styles.wrapperLayoutLoading}>
      <ActivityIndicator size={size} color={color} />
    </View>
  );
};
