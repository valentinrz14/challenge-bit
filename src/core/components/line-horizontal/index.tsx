import React, { FunctionComponent } from 'react';
import { StyleSheet, View } from 'react-native';

/*
 ** Types
 */

interface LineHorizontalProps {
  height: number;
  bgColor: string;
}
/*
 ** Styles
 */
const styles = StyleSheet.create({
  line: {
    width: '100%',
  },
});
/*
 ** Component
 */

export const LineHorizontal: FunctionComponent<LineHorizontalProps> = ({
  bgColor,
  height,
}) => {
  return <View style={[styles.line, { backgroundColor: bgColor, height }]} />;
};
