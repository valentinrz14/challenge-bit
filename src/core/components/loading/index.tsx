import { Colors } from '@core/styles/colors';
import React, { FunctionComponent } from 'react';
import { ActivityIndicator, StyleSheet, View } from 'react-native';

/*
 ** Types
 */
interface LoadingProps {
  color?: string;
  size?: 'large' | 'small';
}
/*
 ** Styles
 */

const styles = StyleSheet.create({
  wrapperLayoutLoading: {
    marginTop: 20,
    alignItems: 'center',
    justifyContent: 'center',
  },
});
/*
 ** Component
 */
export const Loading: FunctionComponent<LoadingProps> = ({
  color = Colors.blue,
  size = 'large',
}) => {
  return (
    <View style={styles.wrapperLayoutLoading}>
      <ActivityIndicator size={size} color={color} />
    </View>
  );
};
