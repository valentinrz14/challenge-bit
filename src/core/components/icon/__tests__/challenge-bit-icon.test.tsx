import React from 'react';
import { ChallengeBitIcon } from '@core/components/icon';
import { render } from '@testing-library/react-native';

describe('ChallengeBitIcon', () => {
  it('should render correctly', () => {
    const { toJSON } = render(
      <ChallengeBitIcon
        iconProps={{ name: 'account', size: 20, color: '#000' }}
      />,
    );
    expect(toJSON).toMatchSnapshot();
  });
});
