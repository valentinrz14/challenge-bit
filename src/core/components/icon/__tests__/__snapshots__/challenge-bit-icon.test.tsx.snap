// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChallengeBitIcon should render correctly 1`] = `[Function]`;
